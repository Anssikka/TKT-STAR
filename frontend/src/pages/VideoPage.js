import React, { useState, useEffect } from 'react'
import videoService from '../services/videos'
import VideoList from '../components/videos/VideoList'
import AddVideoForm from '../components/videos/AddVideoForm'
import PageTitle from '../components/PageTitle'
import PageSubtitle from '../components/PageSubtitle'
import '../styles/RecommendationPage.css'

function VideoPage() {
  const [videos, setVideos] = useState([])

  useEffect(() => {
    getVideos()
  }, [])

  const getVideos = async () => {
    try {
      const savedVideos = await videoService.getVideos()
      setVideos(savedVideos)
    } catch (error) {
      console.log(error)
    }
  }

  const addVideo = async video => {
    try {
      const createdVideo = await videoService.createVideo(video)
      setVideos(videos.concat(createdVideo))
    } catch (error) {
      console.log(error)
    }
  }

  return (
    <main className="recommendation-page">
      <PageTitle title="Video Recommendations" />
      <PageSubtitle title="Add a Video" />
      <AddVideoForm handleSubmit={addVideo} />
      <VideoList title="Saved Videos" videos={videos} />
    </main>
  )
}

export default VideoPage
