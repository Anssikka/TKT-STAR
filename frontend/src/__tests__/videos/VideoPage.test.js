import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render, cleanup, waitForElement } from '@testing-library/react'
import VideoPage from '../../pages/VideoPage'
import axios from 'axios'

afterEach(cleanup)

jest.mock('axios')

describe('<VideoPage />', () => {
  it('has correct elements', async () => {
    axios.get.mockResolvedValue({
      data: [
        {
          id: 1,
          title: 'Test 101',
          url: 'https://example.com',
          tags: [
            { id: 1, title: 'Tag #1' },
            { id: 2, title: 'Tag #2' }
          ]
        }
      ]
    })

    const component = render(<VideoPage />)
    await waitForElement(() => component.findByText('Test 101'))
    expect(component.asFragment()).toMatchSnapshot()
  })
})
