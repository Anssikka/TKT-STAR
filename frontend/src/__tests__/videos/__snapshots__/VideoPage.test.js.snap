// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoPage /> has correct elements 1`] = `
<DocumentFragment>
  <main
    class="recommendation-page"
  >
    <h1
      class="page-title"
    >
      Video Recommendations
    </h1>
    <h3
      class="page-subtitle"
    >
      Add a Video
    </h3>
    <div
      class="video-form-wrapper"
    >
      <form
        class="video-form"
      >
        <label>
          Title:
        </label>
        <input
          data-testid="add-video-title"
          placeholder="Type your video title"
          type="text"
          value=""
        />
        <label>
          URL:
        </label>
        <input
          data-testid="add-video-url"
          placeholder="Type your URL here"
          type="text"
          value=""
        />
        <input
          class="form-submit"
          data-testid="add-video-submit"
          type="submit"
          value="Submit"
        />
      </form>
    </div>
    <div>
      <h3
        class="page-subtitle"
      >
        Saved Videos
      </h3>
      <div
        class="recommendation"
      >
        <h5>
          Test 101
        </h5>
        <a
          href="https://example.com"
          target="_blank"
        >
          https://example.com
        </a>
        <h6
          class="recommendation-tag-title"
        >
          Tags:
        </h6>
        <ul>
          <li>
            Tag #1
          </li>
          <li>
            Tag #2
          </li>
        </ul>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
